app/src/main/java/io/github/boogiemonster1o1/eyeyoureadyforit/command/StatsCommand.java

```diff
@@ -0,0 +1,124 @@
+
+public class StatsCommand {
+
+	public static Mono<?> handle(SlashCommandEvent event) {
+
+		if (event.getOption("server").isPresent()) return handleGuildStatsCommand(event);
+		return handleUserStatsCommand(event);
+
+	}
+
+	private static Mono<?> handleUserStatsCommand(SlashCommandEvent event) {
+		// looking at this makes me want to cry
+
+		Mono<User> userMono = Mono.justOrEmpty(event.getOption("users").get().getOption("user")
+				.flatMap(ApplicationCommandInteractionOption::getValue)
+		).flatMap(ApplicationCommandInteractionOptionValue::asUser)
+				.switchIfEmpty(Mono.just(event.getInteraction().getUser()));
+
+
+
+		return userMono.flatMap(user -> {
+			if (user.isBot()) return event.replyEphemeral("**Statistics are not available for bots!**");
+
+			return StatisticsManager.getUserStats(
+					event.getInteraction().getGuildId().orElseThrow(),
+					user.getId())
+					.defaultIfEmpty(new UserStatistic())
+					.flatMap(statistic -> {
+						EmbedCreateSpec embedSpec = new EmbedCreateSpec()
+								.setTitle(String.format("User Statistics: %s#%s", user.getUsername(), user.getDiscriminator()))
+								.setThumbnail(user.getAvatarUrl())
+								.setColor(Color.of(0, 93, 186))
+								.addField("Correct Answers", App.FORMATTER.format(statistic.getCorrectAnswers()), true)
+								.addField("Wrong Answers", App.FORMATTER.format(statistic.getWrongAnswers()), true)
+								.addField("Hints Used", App.FORMATTER.format(statistic.getHintUses()), true)
+								.addField("Tourneys Won", App.FORMATTER.format(statistic.getGamesWon()), true)
+								.addField("Server Rank", App.FORMATTER.format(statistic.getRank()), true)
+								.setFooter(String.format("User ID: %s", user.getId().asString()), null)
+								.setTimestamp(Instant.now());
+
+						return event.acknowledge().then(
+								event.getInteractionResponse().createFollowupMessage(
+										new WebhookMultipartRequest(WebhookExecuteRequest
+												.builder()
+												.addEmbed(embedSpec.asRequest())
+												.build()
+										)
+								)
+						);
+					});
+		});
+	}
+
+	private static Mono<?> handleGuildStatsCommand(SlashCommandEvent event) {
+		return event.getInteraction().getGuild().flatMap(guild -> StatisticsManager.getGuildStats(guild.getId())
+				.defaultIfEmpty(new GuildStatistic(0, 0))
+				.flatMap(statistic -> {
+					return StatisticsManager.getLeaderboard(guild.getId()).flatMap(board -> {
+						ArrayList<Leaderboard> realBoard = (ArrayList<Leaderboard>) board.stream().filter(e -> e.getId().asLong() != 0).collect(Collectors.toList());
+						String lbString = "Nobody has won yet!";
+						String first = "\n";
+						String second = "\n";
+						String third = "\n";
+
+						// tf is this
+						// send help
+						for(Leaderboard lb : realBoard) {
+							switch(lb.getRank()) {
+								case 1:
+									first = String.format("🥇 - <@%s> - %s wins\n", lb.getId().asString(), App.FORMATTER.format(lb.getGamesWon()));
```

Comment:
please replace emoji with appropriate unicode character sequence (`\uXXXX\uXXXX`)

Reply:
don't really see the point since a) we'll likely never be working in a non-unicode environment and b) the emoji is shorter yet infinitely more descriptive